infra/extfunc-lambda/lambda_function.py

```diff
@@ -0,0 +1,204 @@
+        else:
+            raise RuntimeError('url must start with https://')
+    else:
+        req_host = headers['sf-custom-host']
+        req_path = headers.get('sf-custom-path', '/')
+        req_url = f'https://{req_host}{req_path}'
+
+    req_kwargs = parse_header_dict(headers.get('sf-custom-kwargs'))
+
+    req_headers = {
+        k: v.format(**req_kwargs)
+        for k, v in parse_header_dict(headers.get('sf-custom-headers', '')).items()
+    }
+    if 'sf-custom-basicauth' in headers:
+        basicauth = decrypt_if_encrypted(headers['sf-custom-basicauth'])
+        usr, pwd = basicauth.split(':', 1)
+        auth = decrypt_if_encrypted(usr) + ':' + decrypt_if_encrypted(pwd)
+        req_headers['Authorization'] = b'Basic ' + b64encode(auth.encode())
+
+    # query, nextpage_path, results_path
+    req_qs = headers.get('sf-custom-querystring', '')
+    req_nextpage_path = headers.get('sf-custom-nextpage-path', '')
+    req_results_path = headers.get('sf-custom-results-path', '')
+
+    # load the data
+    body = loads(event['body'])
+    res_data = []
+    for row, *args in body.get('data'):
+        data = []
+
+        if event['path'] == '/https':
+            req_method = headers.get('sf-custom-method', 'get').format(*args).upper()
+            if 'sf-custom-json' in headers:
+                req_data = dumps(
+                    parse_header_dict(headers['sf-custom-json'].format(*args))
+                ).encode()
+            else:
+                req_data = headers.get('sf-custom-data', '').format(*args).encode()
+            next_url = req_url.format(*args)
+            next_url += ('?' + req_qs.format(*args)) if req_qs else ''
+
+            while next_url:
+                req = Request(
+                    next_url, method=req_method, headers=req_headers, data=req_data
+                )
+                links_headers = None
+                try:
+                    response = urlopen(req)
+                    links_headers = parse_header_links(
+                        ','.join(response.headers.get_all('link', []))
+                    )
+                    response_body = response.read()
+                    result = pick(req_results_path, loads(response_body))
+                except HTTPError as e:
+                    result = {'error': f'{e.status} {e.reason}'}
+                except URLError as e:
+                    result = {
+                        'error': f'URLError',
+                        'reason': str(e.reason),
+                        'url': next_url,
+                    }
+                except JSONDecodeError as e:
+                    result = {
+                        'error': 'JSONDecodeError',
+                        'text': response_body.decode(),
+                    }
+
+                if req_nextpage_path and isinstance(result, list):
+                    data += result
+                    nextpage = pick(req_nextpage_path, result)
```

Comment:
when there are too many pages and this runs for >30s, the API Gateway throws an exception. return after 10s here, instead.